esphome/components/http_request/__init__.py

```diff
@@ -137,6 +145,11 @@ async def to_code(config):
 
     if CORE.is_esp32:
         if CORE.using_esp_idf:
+            if buffer_size := config.get(CONF_BUFFER_SIZE_RX):
+                cg.add(var.set_buffer_size_rx(buffer_size))
+            if buffer_size := config.get(CONF_BUFFER_SIZE_TX):
+                cg.add(var.set_buffer_size_tx(buffer_size))
```

Comment:
There will always be a value when using idf because of the default
```suggestion
            cg.add(var.set_buffer_size_rx(config[CONF_BUFFER_SIZE_RX]))
            cg.add(var.set_buffer_size_tx(config[CONF_BUFFER_SIZE_TX]))
```